packages/vue-language-core/src/utils/globalTypes.ts

```diff
@@ -118,6 +118,8 @@ declare function __VLS_componentProps<T, K>(comp: T, fnReturn: K):
 	__VLS_PickNotAny<K, {}> extends { __ctx: { props: infer P } } ? NonNullable<P>
 	: T extends (props: infer P, ...args: any) => any ? NonNullable<P> :
 	{};
+
+declare function __VLS_cast<T>(x: T): T;
```

Comment:
Does that really work? I would imagine that it would cause a type error when passed x doesn't match T. To cast it would need to take unknown argument or something

Reply:
@rchl Ahhhh this is not `casting` some type to another type. This function is created to make the error more reasonable:

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXzCgGcMAeAFQD4AKDALnnIEoHyBuAKA7DxKRxzwAvPADe8KAwCM8AL4SiYiQxIwsqAOZz4Aeh2MAFlkVEDOAO6KMBhCBgwcMeDkTwMATwAOCHqgBudkS4qNy8GPAARrDCBMRk4pLwqupasjQJ0nJMuvrkRiZmlm428HYOTi5uXggAtsY1UBhgBhxAA